test/lib/course_planner/classes/classes_test.exs

```diff
@@ -0,0 +1,119 @@
+defmodule CoursePlanner.ClassesTest do
+  use CoursePlannerWeb.ModelCase
+
+  import CoursePlanner.Factory
+  alias CoursePlanner.Classes
+
+
+  test "sort by starting time when input is an empty list" do
+    assert [] = Classes.sort_by_starting_time([])
+  end
+
+  test "sort by starting time for multiple classes" do
+    class1 = insert(:class, date: Timex.now(), starting_at: Timex.shift(Timex.now(), hours: -2))
+    class2 = insert(:class, date: Timex.now(), starting_at: Timex.shift(Timex.now(), hours: +1))
+    class3 = insert(:class, date: Timex.now(), starting_at: Timex.now())
+    class4 = insert(:class, date: Timex.now(), starting_at: Timex.shift(Timex.now(), minutes: -2))
+    class5 = insert(:class, date: Timex.now(), starting_at: Timex.shift(Timex.now(), minutes: +2))
+
+    classes = [class1, class2, class3, class4, class5]
+    expected_result = [class1, class4, class3, class5, class2]
+
+    assert expected_result == Classes.sort_by_starting_time(classes)
+  end
+
+  test "split past and next classes when input is an empty list" do
+    assert {[],[]} = Classes.split_past_and_next([])
+  end
+
+  test "split past and next classes for multiple classes" do
+    class1 = insert(:class, date: Timex.now(), starting_at: Timex.shift(Timex.now(), hours: -2))
+    class2 = insert(:class, date: Timex.now(), starting_at: Timex.shift(Timex.now(), hours: +1))
+    class3 = insert(:class, date: Timex.now(), starting_at: Timex.now())
+    class4 = insert(:class, date: Timex.now(), starting_at: Timex.shift(Timex.now(), minutes: -2))
+    class5 = insert(:class, date: Timex.now(), starting_at: Timex.shift(Timex.now(), minutes: +2))
+
+    classes = [class1, class2, class3, class4, class5]
+    expected_past_classes =
+      [class1.id, class4.id]
```

Comment:
In the previous test you asserted the whole class struct, but here you're asserting only the id. Was there a problem asserting the whole struct? Would be way less verbose.

Reply:
i check this out , cant remember :P

Reply:
the reason is that in the previous test is testing a sorting function and this test just give past and previous classes so it might not be sorted, then I had to sort them based on id so they are going to be stable.
but now i sort them and check the whole struct. I see the point you are mentioning as well

Reply:
done